src/ui/SearchInterface/FacetValueStateHandler.ts

```diff
@@ -0,0 +1,59 @@
+import { QueryStateModel } from '../../models/QueryStateModel';
+import { Component } from '../Base/Component';
+import { BaseComponent } from '../Base/BaseComponent';
+
+type QueryState = { [key: string]: any };
+type FacetValueState = { [key: string]: any };
+type ComponentsFetcher = (componentId: string) => Component[];
+
+export class FacetValueStateHandler {
+  constructor(private componentsFetcher: ComponentsFetcher) {}
```

Comment:
Here I also think putting the type directly would be better. It avoid having to go back to read the type definition :)